app/pages/project/floating-ips/FloatingIpsPage.tsx

```diff
@@ -0,0 +1,258 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { Link, Outlet, type LoaderFunctionArgs } from 'react-router-dom'
+
+import {
+  apiQueryClient,
+  useApiMutation,
+  useApiQueryClient,
+  usePrefetchedApiQuery,
+  type FloatingIp,
+  type Instance,
+} from '@oxide/api'
+import { useQueryTable, type MenuAction } from '@oxide/table'
+import {
+  buttonStyle,
+  EmptyMessage,
+  IpGlobal24Icon,
+  Listbox,
+  Modal,
+  Networking24Icon,
+  PageHeader,
+  PageTitle,
+  TableActions,
+} from '@oxide/ui'
+
+import { getProjectSelector, useProjectSelector } from 'app/hooks'
+import { confirmDelete } from 'app/stores/confirm-delete'
+import { addToast } from 'app/stores/toast'
+import { pb } from 'app/util/path-builder'
+
+const EmptyState = () => (
+  <EmptyMessage
+    icon={<Networking24Icon />}
+    title="No Floating IPs"
+    body="You need to create a Floating IP to be able to see it here"
+    buttonText="New Floating IP"
+    buttonTo={pb.floatingIpsNew(useProjectSelector())}
+  />
+)
+
+FloatingIpsPage.loader = async ({ params }: LoaderFunctionArgs) => {
+  const { project } = getProjectSelector(params)
+  await Promise.all([
+    apiQueryClient.prefetchQuery('floatingIpList', {
+      query: { project, limit: 25 },
+    }),
+    apiQueryClient.prefetchQuery('instanceList', {
+      query: { project },
+    }),
```

Comment:
I love the list trick, and I want to make followup PRs doing this in other places where we're currently fetching names one by one using the resource ID. I'm not sure the default page size of 100 is enough, though. I think we either want to go big here (like 1000) _or_ get elaborate and fall back to fetching the name by instance ID if it's somehow not in the 100 we fetched. I don't think there will regularly be 1000 instances in projects any time soon, so I am leaning toward keeping it simple and just fetching a lot.